tests/integration/pilot/tunneling/tunneling_test.go

```diff
@@ -0,0 +1,416 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tunneling
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource"
+	kubetest "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/pilot/common"
+)
+
+type tunnelingTestCase struct {
+	// name must be unique, because it's used in the requested URL and then it's searched for in the access.log file,
+	// so a duplicated name would make test false positive
+	name string
+	// protocols specifies what types of requests to test; it can contain "http" or "https" values
+	protocols []string
+	// istioResourcesToApply is a list of files with Istio resources required to apply to test a particular case
+	istioResourcesToApply []string
+}
+
+func testCases() []tunnelingTestCase {
+	return []tunnelingTestCase{
+		{
+			"gateway/tcp",
+			[]string{"http", "https"},
+			[]string{
+				"gateway/tcp/virtual-service.yaml",
+				"gateway/tcp/gateway.yaml",
+			},
+		},
+		//{
+		//	"gateway/tls/istio-mutual",
+		//	[]string{"http", "https"},
+		//	[]string{
+		//		"gateway/tls-istio-mutual/mtls.yaml",
+		//		"gateway/tls-istio-mutual/virtual-service.yaml",
+		//		"gateway/tls-istio-mutual/gateway.yaml",
+		//	},
+		//},
+		{
+			"gateway/tls/passthrough",
+			[]string{"https"},
+			[]string{
+				"gateway/tls-passthrough/virtual-service.yaml",
+				"gateway/tls-passthrough/gateway.yaml",
+			},
+		},
+		{
+			"sidecar",
+			[]string{"http", "https"},
+			[]string{
+				"sidecar/virtual-service.yaml",
+			},
+		},
+	}
+}
+
+var i istio.Instance
+
+const (
+	http1 = "HTTP1"
+	http2 = "HTTP2"
+)
+
+func TestMain(m *testing.M) {
```

Comment:
There is no need to make a whole new suite for this test. Each suite represents substantial cost of redploying all apps+istio.

Reply:
Yes, sure. I created new suite to run only tunnel-related tests and the I forgot to remove it.